test/lib/graph/helpers/createWorkspace.js

```diff
@@ -0,0 +1,270 @@
+		configuration: {
+			specVersion: "workspace/1.0",
+			metadata: {
+				name: "default"
+			},
+			dependencyManagement: {
+				resolutions: [{
+					path: "resolution/path"
+				}]
+			}
+		}
+	}, "Created Workspace instance with correct parameters");
+});
+
+test("createWorkspace: Using invalid object", async (t) => {
+	const {createWorkspace} = t.context;
+
+	const err = await t.throwsAsync(createWorkspace({
+		cwd: "cwd",
+		configObject: {
+			metadata: {
+				name: "default"
+			},
+			dependencyManagement: {
+				resolutions: [{
+					path: "resolution/path"
+				}]
+			}
+		}
+	}));
+	t.true(err.message.includes("Invalid workspace configuration"), "Threw with validation error");
+});
+
+test("createWorkspace: Using file", async (t) => {
+	const {createWorkspace, MockWorkspace, workspaceConstructorStub} = t.context;
+
+	const res = await createWorkspace({
+		cwd: "cwd",
+		name: "default",
+		configPath: path.join(libraryHPath, "ui5-workspace.yaml")
+	});
+
+	t.true(res instanceof MockWorkspace, "Returned instance of Workspace");
+
+	t.is(workspaceConstructorStub.callCount, 1, "Workspace constructor got called once");
+	t.deepEqual(workspaceConstructorStub.getCall(0).args[0], {
+		cwd: libraryHPath,
+		configuration: {
+			specVersion: "workspace/1.0",
+			metadata: {
+				name: "default"
+			},
+			dependencyManagement: {
+				resolutions: [{
+					path: "../library.d"
+				}]
+			}
+		}
+	}, "Created Workspace instance with correct parameters");
+});
+
+test("createWorkspace: Using invalid file", async (t) => {
+	const {createWorkspace} = t.context;
+
+	const err = await t.throwsAsync(createWorkspace({
+		cwd: "cwd",
+		name: "default",
+		configPath: path.join(libraryHPath, "invalid-ui5-workspace.yaml")
+	}));
+
+	t.true(err.message.includes("Invalid workspace configuration"), "Threw with validation error");
+});
+
+test("createWorkspace: Using missing file", async (t) => {
+	const {createWorkspace, workspaceConstructorStub} = t.context;
+
+	const res = await createWorkspace({
+		cwd: path.join(fixturesPath, "library.d"),
+		name: "default",
+		configPath: "ui5-workspace.yaml"
+	});
+
+	t.is(res, null, "Returned no workspace");
+
+	t.is(workspaceConstructorStub.callCount, 0, "Workspace constructor did not get called");
+});
+
+test("createWorkspace: Using missing file and non-default name", async (t) => {
+	const {createWorkspace, workspaceConstructorStub} = t.context;
+
+	const err = await t.throwsAsync(createWorkspace({
+		cwd: path.join(fixturesPath, "library.d"),
+		name: "special",
+		configPath: "ui5-workspace.yaml"
+	}));
+
+	const filePath = path.join(fixturesPath, "library.d", "ui5-workspace.yaml");
+	t.true(err.message.startsWith(
+		`Failed to load workspace configuration from path ${filePath}: `), "Threw with expected error message");
+
+	t.is(workspaceConstructorStub.callCount, 0, "Workspace constructor did not get called");
+});
+
+test("createWorkspace: Using non-default file and non-default name", async (t) => {
```

Comment:
the "special" config is missing in the config. if the "special" would exists, it would work, correct?

Reply:
should we add this to the test name? Its not mentioned anywhere, isn't?

Reply:
Yes, now I get what you mean 👍 
Done and added another test for the positive case where the name matches with the one in the file

Reply:
Done